packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -371,27 +371,32 @@ class FlutterDriverExtension {
     return null;
   }
 
-  /// Returns a future that waits until no pending frame is scheduled (frame is synced).
+  /// Returns a future that waits until the app is idle.
   ///
   /// Specifically, it checks:
-  /// * Whether the count of transient callbacks is zero.
-  /// * Whether there's no pending request for scheduling a new frame.
+  /// * Whether there's no pending [MethodChannel.invokeMethod] calls.
+  /// * Whether the count of transient callbacks is zero, if "frame sync" is enabled.
+  /// * Whether there's no pending request for scheduling a new frame, if "frame sync" is enabled.
   ///
-  /// We consider the frame is synced when both conditions are met.
+  /// We consider the app idle when the conditions listed above are met.
   ///
   /// This method relies on a Flutter Driver mechanism called "frame sync",
   /// which waits for transient animations to finish. Persistent animations will
   /// cause this to wait forever.
   ///
   /// If a test needs to interact with the app while animations are running, it
-  /// should avoid this method and instead disable the frame sync using
-  /// `set_frame_sync` method. See [FlutterDriver.runUnsynchronized] for more
-  /// details on how to do this. Note, disabling frame sync will require the
-  /// test author to use some other method to avoid flakiness.
-  Future<Result> _waitUntilNoPendingFrame(Command command) async {
-    await _waitUntilFrame(() {
-      return SchedulerBinding.instance.transientCallbackCount == 0
-          && !SchedulerBinding.instance.hasScheduledFrame;
+  /// should instead disable the frame sync using the `set_frame_sync` method.
+  /// See [FlutterDriver.runUnsynchronized] for more details on how to do this.
+  /// Note, disabling frame sync will require the test author to use some other method to avoid flakiness.
+  Future<Result> _waitUntilIdle(Command command) async {
+    await _waitUntilCondition(() {
```

Comment:
Why's _waitUntilFrame not good enough here?

Reply:
Ah, is it because a reduction in pendingChannelInvokeMethodCount may not schedule a frame?

Reply:
Ya... absolutely. At the same time, we might want to wait for all the conditions to be true in one single check (instead of different futures).

Reply:
Is not having a frame scheduled after a method channel call returns something we think will really happen?  I'm trying to think of a use case for such behavior.  It'd generally be a fire-and-forget call (maybe logging or analytics?).

> we might want to wait for all the conditions to be true in one single check

We can accomplish that with a post-frame callback condition in `_waitUntilFrame()`